config/src/main/java/com/typesafe/config/impl/SerializedConfigValue.java

```diff
@@ -404,8 +434,9 @@ private static AbstractConfigValue readValue(DataInput in, SimpleConfigOrigin ba
             } else if (code == SerializedField.VALUE_DATA) {
                 if (origin == null)
                     throw new IOException("Origin must be stored before value data");
-                in.readInt(); // discard length
-                value = readValueData(in, origin);
+                int byteLength = in.readInt()*2; //The byte length of the String to be read. Multiplied by 2 is because one char takes 2 bytes generally.
```

Comment:
The real byte length is returned by `in.readInt()` (see writeField function). I think, it's better to put all the calculations into readValueData function.